controller/cmd/service-mirror/cluster_watcher.go

```diff
@@ -405,6 +438,16 @@ func (rcsw *RemoteClusterServiceWatcher) handleRemoteServiceUpdated(ev *RemoteSe
 	serviceInfo := fmt.Sprintf("%s/%s", ev.remoteUpdate.Namespace, ev.remoteUpdate.Name)
 	rcsw.log.Debugf("Updating remote mirrored service %s/%s", ev.localService.Namespace, ev.localService.Name)
 
+	if ev.localEndpoints.Labels[consts.RemoteGatewayNameLabel] != ev.gatewayData.Name {
+		rcsw.probeChan <- &MirroredServiceUnpaired{
+			serviceName:      ev.localService.Name,
+			serviceNamespace: ev.localService.Namespace,
+			gatewayName:      ev.localEndpoints.Labels[consts.RemoteGatewayNameLabel],
+			gatewayNs:        ev.localEndpoints.Labels[consts.RemoteGatewayNsLabel],
+			clusterName:      rcsw.clusterName,
+		}
+	}
+
```

Comment:
When the remote service gets updated, you're issuing a `MirroredServiceUnpaired` event. Isn't a following `MirroredServicePaired` event required?

Reply:
Yes.. that makes sense. For reference if the gateway for a particular service has changed to another gateway, we need to unpair and then pair again, potentially resulting in a new probe being created.